history-network.md

```diff
@@ -60,10 +60,19 @@ Each of these data items are represented as a key-value pair. Denote the key for
 
 All `content` items are transmitted as RLP-encoded byte arrays.
 
-We derive a `content-id` from the `content-key` as `H(content-key)` where `H` denotes the SHA-256 hash function, which outputs 32-byte values. The `content-id` represents the key in the DHT that we use for `distance` calculations.
-
 All `content-key` values are encoded and decoded according to SSZ sedes.
 
+In addition to the `content-key`, each piece of `content` has a corresponding `content-id`. The `content-id` is the 32 byte value that we use (along with the `node-id`) for `distance` calculations.
+
+We derive a `content-id` from the `content-key` as follows:
+
+  1.) If the `content-key` has `content-type` 0x04, change the `content-type` to 0x02 and remove the `transaction-hash` field.
+  2.) Calculate `H(content-key)` where `H` denotes the SHA-256 hash function.
+
+Step 1 is done so that a `content-key` for a single transaction maps to the same `content-id` as the entire block body that contains the desired transaction. The individual transaction can then be retrieved and sent over the wire by itself. Otherwise the network would need to store each transaction twice: individually, as well as within the block body. This, along with the required inclusion proofs, would roughly triple the required network storage space.
```

Comment:
If the content-key for transactions in a block body and the content-key for the block body itself map to same content-id, how can the individual transaction be retrieved, if we are using `content-id` as the key into db? Then there is no unique mapping, so it might be useful to clarify that. Otherwise, I think it looks good.

Reply:
Yea I should probably add something like: "Keep the transaction-hash in memory, because it will be necessary for getting the desired transaction out of the block that is returned from the DB"